apps/system/utils/utils_prodconfig.c

```diff
@@ -0,0 +1,72 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <errno.h>
+#include <stdio.h>
+#include <fcntl.h>
+
+#include <tinyara/prodconfig.h>
+#include <tinyara/fs/ioctl.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name : utils_prodconfig
+ *
+ * Description:
+ *   It reads the hw configurations from prodconfig drivers
+ *   and prints the status like swd enable/disable.
+ *
+ ****************************************************************************/
+int utils_prodconfig(int argc, char **args)
+{
+	int prodconfig_fd;
+	int ret;
+	prodconfig_fd = open(PRODCONFIG_DRVPATH, O_RDWR);
+	if (prodconfig_fd < 0) {
+		printf("Fail to open prodconfig, errno %d.\n", get_errno());
+		return ERROR;
+	}
+
+	ret = ioctl(prodconfig_fd, PRODCONFIGIOC_SWD, 0);
+	if (ret == SWD_ENABLED) {
+		printf("HW Debug : On\n");
+	} else if (ret == SWD_DISABLED) {
+		printf("HW Debug : Off\n");
+	} else {
+		printf("Fail to check swd config.\n");
+	}
+
+	ret = ioctl(prodconfig_fd, PRODCONFIGIOC_DOWNLOAD, 0);
+	if (ret == DOWNLOAD_ENABLED) {
+		printf("Download : Downloader & Hotfix Available\n");
+	} else if (ret == DOWNLOAD_DISABLED) {
+		printf("Download : Hotfix Only Available\n");
+	} else {
+		printf("Fail to check download config.\n");
+	}
```

Comment:
They will be used only in this util so that we can use `read` operation instead of `ioctl`. If we use the read, no need additional definitions in ioctl.